### PR TITLE
Added type code to participant and set when event is generated. Feature ...

### PR DESCRIPTION
**Pull request by jacobvanorder**
...#3599

### CODE REVIEW

File: NCSNavField/Participant.h [hunk 1]
```diff
@@ -14,6 +14,7 @@
 @interface Participant : NSManagedObject
 
 @property (nonatomic, retain) NSString * pId;
+@property (nonatomic, assign) int typeCode;
```
Comment (jdzak) [inline NCSNavField/Participant.h]: Can you switch from int to NSNumber?

Comment (jacobvanorder) [inline NCSNavField/Participant.h]: Fixed.

File: NCSFieldConstants.h [hunk 2]
```diff
@@ -40,3 +40,8 @@
 FOUNDATION_EXPORT NSString * EVENT_TEMPLATE_PBS_ELIGIBILITY_LEGACY;
 FOUNDATION_EXPORT NSString * EVENT_TEMPLATE_PREG_VISIT_ONE_LEGACY;
 
+// Participant Type Code
+
+FOUNDATION_EXPORT int PARTICIPANT_TYPE_PBS_SCREENING;
+FOUNDATION_EXPORT int PARTICIPANT_TYPE_BIRTH_COHORT;
```
Comment (jdzak) [inline NCSFieldConstants.h]: Can you change these participant type constants to NSNumber also?

Comment (jacobvanorder) [inline NCSFieldConstants.h]: I tried but it gave me this: 

Screen Shot 2013-04-26 at 11 49 27 AM

File: NCSNavField/main.xcdatamodeld/20130424144006_participant_typeCode.xcdatamodel/contents [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    <entity name="NUQuestion" representedClassName="NUQuestion" syncable="YES">
+        <attribute name="referenceIdentifier" optional="YES" attributeType="String" indexed="YES" syncable="YES"/>
+        <attribute name="text" optional="YES" attributeType="String" syncable="YES"/>
+        <attribute name="uuid" optional="YES" attributeType="String" indexed="YES" syncable="YES"/>
+        <relationship name="answers" optional="YES" toMany="YES" deletionRule="Nullify" ordered="YES" destinationEntity="NUAnswer" inverseName="question" inverseEntity="NUAnswer" syncable="YES"/>
+        <relationship name="instrumentTemplate" optional="YES" minCount="1" maxCount="1" deletionRule="Nullify" destinationEntity="InstrumentTemplate" inverseName="questions" inverseEntity="InstrumentTemplate" syncable="YES"/>
+    </entity>
+    <entity name="Participant" representedClassName="Participant" syncable="YES">
+        <attribute name="pId" optional="YES" attributeType="String" syncable="YES"/>
+        <attribute name="typeCode" optional="YES" attributeType="Integer 32" defaultValueString="0" syncable="YES"/>
```
Comment (jdzak) [inline NCSNavField/main.xcdatamodeld/20130424144006_participant_typeCode.xcdatamodel/contents]: Sorry, one more thing.. can you remove the default value?

Comment (jacobvanorder) [inline NCSNavField/main.xcdatamodeld/20130424144006_participant_typeCode.xcdatamodel/contents]: No problem. I think that's the default but will check for it in the future.

**Pull request merged**